tests/system/Database/Live/PreparedQueryTest.php

```diff
@@ -162,12 +166,20 @@ public function testDeallocatePreparedQuery()
 
         $this->query->close();
 
+        // Try to execute a non-existing prepared statement
         try {
             $this->query->execute('bar', 'bar@example.com', 'GB');
         } catch (BadMethodCallException $e) {
             $this->assertSame('You must call prepare before trying to execute a prepared statement.', $e->getMessage());
         }
 
+        // Try to close a non-existing prepared statement
+        try {
+            $this->query->close();
+        } catch (BadMethodCallException $e) {
+            $this->assertSame('Cannot call close on a non-existing prepared statement.', $e->getMessage());
+        }
```

Comment:
Please write tests for Exceptions using `$this->expectException()` and `$this->expectExceptionMessage()`.
Not try catch.

Reply:
Done, I've refactored the corresponding unit tests